src/include/miopen/nllloss/problem_description.hpp

```diff
@@ -0,0 +1,222 @@
+/*******************************************************************************
+ *
+ * MIT License
+ *
+ * Copyright (c) 2024 Advanced Micro Devices, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ *******************************************************************************/
+
+#pragma once
+
+#include <miopen/problem_description_base.hpp>
+#include <miopen/activ.hpp>
+#include <miopen/tensor.hpp>
+#include <cassert>
+#include <string>
+
+namespace miopen {
+
+struct NetworkConfig;
+
+namespace nllloss {
+
+struct ProblemDescription : ProblemDescriptionBase
+{
+    ProblemDescription(const TensorDescriptor& inputDesc_,
+                       const TensorDescriptor& targetDesc_,
+                       const TensorDescriptor& weightDesc_,
+                       const TensorDescriptor& outputDesc_,
+                       int32_t ignore_index_,
+                       bool is_fwd_)
+        : inputDesc(inputDesc_),
+          targetDesc(targetDesc_),
+          weightDesc(weightDesc_),
+          outputDesc(outputDesc_),
+          ignore_index(ignore_index_),
+          is_fwd(is_fwd_)
+    {
+    }
+
+    const TensorDescriptor& GetInputDesc() const { return inputDesc; }
+    const TensorDescriptor& GetTargetDesc() const { return targetDesc; }
+    const TensorDescriptor& GetWeightDesc() const { return weightDesc; }
+    const TensorDescriptor& GetOutputDesc() const { return outputDesc; }
+    int32_t GetIgnoreIndex() const { return ignore_index; }
+
+    bool IsValidLength() const
+    {
+        if(targetDesc.GetLengths()[0] != inputDesc.GetLengths()[0])
+#if MIOPEN_BUILD_DEV || !MIOPEN_NDEBUG
+            MIOPEN_THROW(miopenStatusBadParm, "NLLLoss: Tensor sizes do not match.");
+#else
```

Comment:
you can remove return false. Since the input is not valid, it just throws.

Reply:
Do you mean that we should throw every time, not just in MIOPEN_BUILD_DEV or !MIOPEN_NDEBUG situation?

Reply:
Yes, the if statement that checks for not valid throws regardless of (MIOPEN_BUILD_DEV || !MIOPEN_NDEBUG).
If it has nothing to do with the valid, just return false.

Reply:
Got it, I applied all your comments in this PR to the other branch that is implementing additional 
2d and 5d (as modnn). Should I also apply them here. By the way, please keep reviewing.